internal/controllers/historicalv2.go

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/DIMO-Network/device-data-api/internal/services/elastic"
+	"github.com/DIMO-Network/shared/middleware/privilegetoken"
+	"github.com/gofiber/fiber/v2"
+)
+
+// GetHistoricalPermissionedV2 godoc
+// @Description  Get all historical data for a tokenID, within start and end range
+// @Tags         device-data
+// @Produce      json
+// @Success      200
+// @Param        tokenID  path   int64  true   "token id"
+// @Param        startTime     query  string  false  "startTime is an RFC3339 formatted date-time string. if empty two weeks back"
+// @Param        endTime       query  string  false  "endTime is an RFC3339 formatted date-time string. if empty today"
+// @Param        buckets       query  string  false  "number of data points to return, default 1000"
+// @Security     BearerAuth
+// @Router       /v2/vehicle/{tokenID}/history [get]
+func (d *DeviceDataController) GetHistoricalPermissionedV2(c *fiber.Ctx) error {
+	// get the startTime and endTime from the query parameters and convert them to time.Time
+	var err error
+	startTime, err := getTimeFromQuery(c, "startTime")
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+	endTime, err := getTimeFromQuery(c, "endTime")
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+
+	buckets, err := strconv.ParseInt(c.Query("buckets", "0"), 10, 64)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("error parsing buckets value %q: %v", c.Query("buckets"), err))
+	}
+
+	// get vehicleID from tokenID
+	tokenIDParam := c.Params("tokenID")
+	tokenID, err := strconv.ParseInt(tokenIDParam, 10, 64)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("error parsing tokenID: %v", err))
+	}
+	ctx, cancel := context.WithTimeout(c.Context(), defaultTimeout)
+	defer cancel()
+	userDevice, err := d.deviceAPI.GetUserDeviceByTokenID(ctx, tokenID)
+	if err != nil {
+		return fmt.Errorf("failed to get user device by token id: %w", err)
+	}
+
+	// get the token claims from the context
+	claims, ok := c.Locals("tokenClaims").(privilegetoken.CustomClaims)
+	if !ok {
+		return fiber.NewError(fiber.StatusUnauthorized, "no token claims found")
+	}
+
+	params := elastic.GetHistoryParams{
+		DeviceID:     userDevice.GetId(),
+		StartTime:    startTime,
+		EndTime:      endTime,
+		Buckets:      int(buckets),
+		PrivilegeIDs: claims.PrivilegeIDs,
+	}
+
+	result, err := d.esService.GetHistory(ctx, params)
+	if err != nil {
+		return fmt.Errorf("failed to get history: %w", err)
+	}
+	resp := historyResp{Status: result}
```

Comment:
Hum, was it always `"status"` singular? That seems off.

Reply:
It was before, but its v2 so now would be the time to update it https://github.com/DIMO-Network/device-data-api/blob/473e4673e2db8c27b160c6a35d466b3634909a28/internal/controllers/device_data_controller.go#L396

Reply:
Let's change it now.